src/borg/archiver.py

```diff
@@ -696,6 +698,195 @@ def peek_and_store_hardlink_masters(item, matched):
 
     @with_repository()
     @with_archive
+    def do_export_tar(self, args, repository, manifest, key, archive):
+        """Export archive contents as a tarball"""
+        self.output_list = args.output_list
+
+        filter = None
+        if args.tar_filter == 'auto':
+            # Note that filter remains None if tarfile is '-'.
+            if args.tarfile.endswith('.tar.gz'):
+                filter = 'gzip'
+            elif args.tarfile.endswith('.tar.bz2'):
+                filter = 'bzip2'
+            elif args.tarfile.endswith('.tar.xz'):
+                filter = 'xz'
```

Comment:
I believe we did about create-from-tar which uses/d the same design. The reason stays the same, but is even more important (since compressors are slower): That way the compressor inherently runs in a separate thread and does not hog our main thread.